ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt

```diff
@@ -16,60 +16,28 @@
 package com.google.android.ground.ui.datacollection.tasks.date
 
 import android.app.DatePickerDialog
-import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.core.view.doOnAttach
-import androidx.fragment.app.activityViewModels
-import com.google.android.ground.BR
 import com.google.android.ground.databinding.DateTaskFragBinding
-import com.google.android.ground.ui.common.AbstractFragment
-import com.google.android.ground.ui.datacollection.DataCollectionViewModel
-import com.google.android.ground.ui.datacollection.tasks.TaskFragment
+import com.google.android.ground.ui.datacollection.components.TaskView
+import com.google.android.ground.ui.datacollection.components.TaskViewWithHeader
+import com.google.android.ground.ui.datacollection.tasks.AbstractTaskFragment
 import dagger.hilt.android.AndroidEntryPoint
 import java.util.*
-import kotlin.properties.Delegates
 
 @AndroidEntryPoint
-class DateTaskFragment : AbstractFragment(), TaskFragment<DateTaskViewModel> {
-  private val dataCollectionViewModel: DataCollectionViewModel by activityViewModels()
-  override lateinit var viewModel: DateTaskViewModel
-  override var position by Delegates.notNull<Int>()
-  private lateinit var binding: DateTaskFragBinding
+class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    if (savedInstanceState != null) {
-      position = savedInstanceState.getInt(TaskFragment.POSITION)
-    }
-  }
-
-  override fun onSaveInstanceState(outState: Bundle) {
-    super.onSaveInstanceState(outState)
-    outState.putInt(TaskFragment.POSITION, position)
-  }
-
-  override fun onCreateView(
-    inflater: LayoutInflater,
-    container: ViewGroup?,
-    savedInstanceState: Bundle?
-  ): View {
-    super.onCreateView(inflater, container, savedInstanceState)
-    binding = DateTaskFragBinding.inflate(inflater, container, false)
-
-    return binding.root
-  }
-
-  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-    super.onViewCreated(view, savedInstanceState)
+  override fun onCreateTaskView(inflater: LayoutInflater, container: ViewGroup?): TaskView =
+    TaskViewWithHeader.create(inflater)
 
-    view.doOnAttach {
-      viewModel = dataCollectionViewModel.getTaskViewModel(position) as DateTaskViewModel
-      binding.lifecycleOwner = this
-      binding.setVariable(BR.viewModel, viewModel)
-      binding.setVariable(BR.fragment, this)
-    }
```

Comment:
The work that is being done here and in other task fragments onAttach needs to continue being done onAttach. This PR currently moves this work from onAttach to onViewCreated, but I believe that will cause the DataCollectionFragmentTest to fail.

Reply:
Resolved.